tests/Kiota.Builder.Tests/Writers/CSharp/CodeMethodWriterTests.cs

```diff
@@ -540,7 +540,7 @@ public void WritesModelFactoryBodyForUnionModels()
         Assert.Contains("ComplexType2Value = complexType2ValueValue", result);
         Assert.Contains("return result", result);
         AssertExtensions.Before("GetStringValue() is string stringValueValue", "GetCollectionOfObjectValues<ComplexType2>", result);
-        AssertExtensions.CurlyBracesAreClosed(result);
+        AssertExtensions.CurlyBracesAreClosed(result, 1);
```

Comment:
what's the rationale for offsetting the curly braces here?

Reply:
The conditional compilation introduces two opening braces for one closing brace. One for each target we compile for.

Reply:
thanks for the clarification here, I missed that aspect.